src/runner.rs

```diff
@@ -400,26 +401,31 @@ pub fn check_test_runner() {
 }
 
 pub trait Termination {
-    fn is_success(&self) -> bool;
+    type Error: fmt::Debug;
+    fn error_value(&self) -> Option<&Self::Error>;
 }
 
 impl Termination for () {
-    fn is_success(&self) -> bool {
-        true
+    type Error = ();
+    fn error_value(&self) -> Option<&()> {
+        None
     }
 }
 
-impl<T, E> Termination for Result<T, E> {
-    fn is_success(&self) -> bool {
-        self.is_ok()
+impl<T, E: fmt::Debug> Termination for Result<T, E> {
```

Comment:
This modifies the `Termination` trait to require `E: fmt::Debug`. My guess is that this should be fine because that's already required of `#[test]`s, but I don't actually know how test systems work.

Reply:
I was modeling it after `std::process::Termination` (which is currently unstable): https://doc.rust-lang.org/src/std/process.rs.html#1644

Can we just copy-paste what they do? I think, I missed the part where they printt the error: https://doc.rust-lang.org/src/std/process.rs.html#1679

Reply:
(with the hope that one day I just switch to `std::process::Termination`)